jujugui/static/gui/src/app/components/entity-details/content/resources/resources.js

```diff
@@ -23,39 +23,77 @@ YUI.add('entity-resources', function() {
   juju.components.EntityResources = React.createClass({
 
     propTypes: {
+      apiUrl: React.PropTypes.string.isRequired,
+      entityId: React.PropTypes.string.isRequired,
       pluralize: React.PropTypes.func.isRequired,
       resources: React.PropTypes.array
     },
 
+    getInitialState: function() {
+      const props = this.props;
+      return {
+        baseUrl: `${props.apiUrl}/${props.entityId}/resource`
+      };
+    },
+
+    /**
+      Generate a single resource to display.
+
+      @param {Object} resource the resource being displayed.
+      @returns {Object} The resource markup.
+    */
+    _generateResource: function(resource) {
+      const revision = resource.Revision;
+      const name = resource.Name;
+      // Get the file extension.
+      const parts = resource.Path.split('.');
+      let extension = '';
+      // If there is a file extension then format it for display.
+      if (parts.length > 1) {
+        extension = `(.${parts.pop()})`;
+      }
+      let itemContent = null;
+      // A revision of 0 should not be considered false.
+      if (revision !== undefined) {
```

Comment:
You can create a charm with resource but not attach the content yet, and I think you get -1 as a revision in that case. As soon as a resource is attached it gets a revision and is downloadable. 
That if I remember correctly.